app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/ConnectivityAlertsViewModel.java

```diff
@@ -19,90 +28,111 @@
 
     private final DisplayGpsConnectivityStatusInteractorFactory mGpsFactory;
     private final DisplayDataConnectivityStatusInteractorFactory mDataFactory;
+    private final Display3GConnectivityStatusInteractorFactory m3GFactory;
+
+    private final String mGpsAlert;
+    private final String mDataAlert;
+    private final String m3GAlert;
+
+    private final Set<String> mDisplayedAlerts;
+
     private ConnectivityAlertsDisplayer mConnectivityAlertsDisplayer;
 
     private DisplayGpsConnectivityStatusInteractor mGpsStatusInteractor;
     private DisplayDataConnectivityStatusInteractor mDataStatusInteractor;
+    private Display3GConnectivityStatusInteractor m3GInteractor;
 
     @Inject
-    LocationFetcher mLocationFetcher;
-
-    @Inject
-    public ConnectivityAlertsViewModel(DisplayGpsConnectivityStatusInteractorFactory gpsFactory,
-                                       DisplayDataConnectivityStatusInteractorFactory dataFactory) {
+    public ConnectivityAlertsViewModel(Context context,
+                                       DisplayGpsConnectivityStatusInteractorFactory gpsFactory,
+                                       DisplayDataConnectivityStatusInteractorFactory dataFactory,
+                                       Display3GConnectivityStatusInteractorFactory threeGFactory) {
         mGpsFactory = gpsFactory;
         mDataFactory = dataFactory;
+        m3GFactory = threeGFactory;
+        mGpsAlert = context.getString(R.string.no_gps_signal);
+        mDataAlert = context.getString(R.string.no_network);
+        m3GAlert = context.getString(R.string.no_3g_network);
+        mDisplayedAlerts = new HashSet<>();
     }
 
-    public void setAlertsDisplayer(ConnectivityAlertsDisplayer connectivityAlertsDisplayer){
+    public void setAlertsDisplayer(ConnectivityAlertsDisplayer connectivityAlertsDisplayer) {
         mConnectivityAlertsDisplayer = connectivityAlertsDisplayer;
     }
 
     public void start() {
-        initGpsStatus();
         startGpsAlerts();
         startDataAlerts();
+        start3GAlerts();
```

Comment:
Absolutely a matter of personal preference but i'd use two methods:
createInteractors()
executeInteractors()